compact.js

```diff
@@ -15,7 +15,7 @@ function compact(array) {
   let resIndex = 0
   const result = []
 
-  if (array == null) {
+  if (array === null) {
```

Comment:
`==` is desirable and most likely intentional because it will also match `undefined`